docs/core-concepts/libraries/libraries-with-cli.md

```diff
@@ -331,154 +318,217 @@ using System.Threading.Tasks;
 
-```javascript
+The `project.json` files for both **AwesomeLibrary.CSharp** and **AwesomeLibrary.FSharp** now need to reference **AwesomeLibrary.Core**.  If you aren't multitargeting, you can use the global `dependencies` entry:
+
+```json
 {
-    "dependencies":{
-        "MyLibrary.Core":""
-        ...
+    "dependnecies":{
+        "AwesomeLibrary.Core":"1.0.0"
     }
-    ...
 }
 ```
 
-To build it all, you would open a command line at the root of the project (`/MyLibrary`).
+If you are multitargeting, you may not be able to use a global `dependencies` entry and may have to reference **AwesomeLibrary.Core** in a target-level `dependencies` entry.  For example, if you were targeting `netstandard1.5`, you could do so like this:
+
+```json
+{
+    "frameworks":{
+        "netstandard1.5":{
+            "dependencies":{
+                "AwesomeLibrary.Core":"1.0.0"
+            }
+        }
+    }
+}
+```
+
+And that's it!  To learn more about `project.json` and all of the configuration you can do, check out the [.NET Core project system](../../project-model/overview.md).
+
+### Structuring a Solution
+
+Another important aspect of multi-project solutions is establishing a good overall project structure.
 
-Using DNX:
+There are many ways to structure projects in .NET, and you have significant freedom in the particular convention that suits your needs.  Here are two approaches you might consider using.
+
+A first approach to structuring this library is to place each project under top-level `/src` and `/test` folders:
 
 ```
-$ dnu restore
-$ cd ../MyLibrary.CSharp
-$ dnu build
-$ cd ../MyLibrary.FSharp
-$ dnu build
+/AwesomeLibrary
+|__global.json
+|__/src
+   |__/AwesomeLibrary.Core
+      |__Source Files
+      |__project.json
+   |__/AwesomeLibrary.CSharp
+      |__Source Files
+      |__project.json
+   |__/AwesomeLibrary.FSharp
+      |__Source Files
+      |__project.json
+/test
+   |__/AwesomeLibrary.Core.Tests
+      |__Test Files
+      |__project.json
+   |__/AwesomeLibrary.CSharp.Tests
+      |__Test Files
+      |__project.json
+   |__/AwesomeLibrary.FSharp.Tests
+      |__Test Files
+      |__project.json
+```
+
+The `global.json` file for this solution would look like this:
+
+```json
+{
+    "projects":[ "src", "test"]
+}
 ```
-    
-Using .NET CLI:
-   
+
+This way follows the pattern current project templates in the `dotnet new` command establish, where all projects are placed under a `/src` directory and all tests are placed under a `/tests` directory.
+
+Here's how you would restore packages and build the entire project:
+
 ```
+$ cd AwesomeLibrary/src
 $ dotnet restore
-$ cd ../MyLibrary.CSharp
+$ cd AwesomeLibrary.FSharp
 $ dotnet build
-$ cd ../MyLibrary.FSharp
+$ cd AwesomeLibrary.CSharp
 $ dotnet build
 ```
```

Comment:
> I can't do that right now. We don't have a .NET CLI-compatible testing framework that customers can use at this time.

@cartermp we **do** have support for xUnit, we are using it in the CLI repos. Not sure what you mean by  this comment.